httptrace/client.go

```diff
@@ -157,7 +158,7 @@ func (t *Transport) RoundTrip(req *http.Request) (*http.Response, error) {
 		e.Response.StatusCode = -1
 	}
 	child.Event(e)
-
+	t.saveCloneRequest(original, nil)
```

Comment:
It would be better to make this a `defer t.saveCloneRequest(original, nil)` directly after the first call to this function. Important if e.g. `RoundTrip(req)` panics and would cause the map to grow unbounded.